ios/NotifeeCore/NotifeeCoreUtil.m

```diff
@@ -572,17 +572,59 @@ + (NSNumber *)convertToTimestamp:(NSDate *)date {
 }
 
 /**
- * Parse UNNotificationRequest to NSDictionary
+ * Parse UNNotificationRequest to NSMutableDictionary
+ * Used by `getDeliveredNotification`
  *
  * @param request UNNotificationRequest
  */
-+ (NSDictionary *)parseUNNotificationRequest:(UNNotificationRequest *)request {
++ (NSMutableDictionary *)parseUNNotificationRequest:(UNNotificationRequest *)request {
   NSMutableDictionary *dictionary = [NSMutableDictionary dictionary];
-  NSMutableDictionary *iosDict = [NSMutableDictionary dictionary];
 
+  dictionary = [self parseUNNotificationContent:request.content];
   dictionary[@"id"] = request.identifier;
 
-  UNNotificationContent *content = request.content;
+  NSDictionary *userInfo = request.content.userInfo;
+
+  // Check for remote details
+  if ([request.trigger isKindOfClass:[UNPushNotificationTrigger class]]) {
+    NSMutableDictionary *remote = [NSMutableDictionary dictionary];
+
+    remote[@"messageId"] = userInfo[@"gcm.message_id"];
+    remote[@"senderId"] = userInfo[@"google.c.sender.id"];
+
+    if (userInfo[@"aps"] != nil) {
+      remote[@"mutableContent"] = userInfo[@"aps"][@"mutable-content"];
+      remote[@"contentAvailable"] = userInfo[@"aps"][@"content-available"];
+    }
+
+    dictionary[@"remote"] = remote;
+  }
+
+  dictionary[@"data"] = [self parseDataFromUserInfo:userInfo];
+
+  return dictionary;
+}
+
++ (NSMutableDictionary *)parseDataFromUserInfo:(NSDictionary *)userInfo {
+  NSMutableDictionary *data = [[NSMutableDictionary alloc] init];
+  for (id key in userInfo) {
+    // build data dict from remaining keys but skip keys that shouldn't be included in data
+    if ([key isEqualToString:@"aps"] || [key hasPrefix:@"gcm."] || [key hasPrefix:@"google."] ||
+        // notifee or notifee_options
+        [key hasPrefix:@"notifee"] ||
+        // fcm_options
+        [key hasPrefix:@"fcm"]) {
```

Comment:
structurally maybe better to have all the magic string constants "mutable-content"/content-available etc etc above plus these ones defined as constants, and separately have an array of "reserved prefixes" then here roll through the reserved prefixes array. Doesn't look too bad right now as they all seem to be referenced once but string constants have a habit of multiplying around